subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/NamedDomainObjectCollectionExtensions.kt

```diff
@@ -366,13 +366,13 @@ fun <T : Any, U : NamedDomainObjectCollection<T>> U.getting(configuration: T.()
 /**
  * Enables typed access to container elements via delegated properties.
  */
-class NamedDomainObjectCollectionDelegateProvider<T>
+class NamedDomainObjectCollectionDelegateProvider<T : Any>
```

Comment:
❓ are these new `T : Any` constraints required by a newer Kotlin version?

Reply:
Yes. Kotlin 1.4.20 reports a warning. See the commit messages